prefect_utils/snowflake.py

```diff
@@ -194,3 +194,140 @@ def load_ga_data_to_snowflake(
+      s3_bucket (str): S3 bucket from which to load data from.
+      s3_key_path (str): Path of the S3 objects relative to `s3_bucket`.
+      sf_credentials (dict): Snowflake public key credentials in the format required by create_snowflake_connection.
+      sf_database (str): Name of the destination database.
+      sf_schema (str): Name of the destination schema.
+      sf_table (str): Name of the destination table.
+      sf_role (str): Name of the snowflake role to assume.
+      sf_warehouse (str): Name of the Snowflake warehouse to be used for loading.
+      sf_storage_integration_name (str): Name of the Snowflake storage integration to use. These are
+              configured in Terraform.
+      sf_file_format (str, optional): Snowflake file format for the Stage. Defaults to 'JSON'.
+      pattern (str, optional): Path pattern/regex to match S3 objects to copy. Defaults to `.*`.
+      overwrite (bool, optional): Whether to overwrite existing data for the given date. Defaults to `False`.
+    """
+
+    sf_connection = create_snowflake_connection(sf_credentials, sf_role, warehouse=sf_warehouse)
+
+    # Check for data existence for this date
+    try:
+        query = """
+        SELECT 1 FROM {table}
+        WHERE date(PROPERTIES:{date_property})='{date}'
+        """.format(
+            table=qualified_table_name(sf_database, sf_schema, sf_table),
+            date=date,
+            date_property=date_property,
+        )
+        cursor = sf_connection.cursor()
+        cursor.execute(query)
+        row = cursor.fetchone()
+    except snowflake.connector.ProgrammingError as e:
+        if "does not exist" in e.msg:
+            # If so then the query failed because the table doesn't exist.
+            row = None
+        else:
+            raise
+
+    if row and not overwrite:
+        return
+
+    try:
+        # Create the generic loading table
+        query = """
+        CREATE TABLE IF NOT EXISTS {table} (
+            ID NUMBER AUTOINCREMENT START 1 INCREMENT 1,
+            LOAD_TIME TIMESTAMP_LTZ DEFAULT CURRENT_TIMESTAMP(),
+            ORIGIN_FILE_NAME VARCHAR(16777216),
+            ORIGIN_FILE_LINE NUMBER(38,0),
+            ORIGIN_STR VARCHAR(16777216),
+            PROPERTIES VARIANT
+        );
+        """.format(
+            table=qualified_table_name(sf_database, sf_schema, sf_table)
+        )
+
+        sf_connection.cursor().execute(query)
+
+        # Delete existing data in case of overwrite.
+        if overwrite:
+            query = """
+            DELETE FROM {table}
+            WHERE date(PROPERTIES:{date_property})='{date}'
+            """.format(
+                table=qualified_table_name(sf_database, sf_schema, sf_table),
+                date=date,
+                date_property=date_property,
+            )
+            sf_connection.cursor().execute(query)
+
+        stage_url = 's3://{bucket}/{path}'.format(bucket=s3_bucket, path=s3_key_path)
```

Comment:
If we want to bulk import instead of mapping, we could change the stage to specify only the bucket, and then in the COPY command we can pass in FILES(s3 paths excluding the bucket) and it should load everything in one go. 

That will also require us to delete rows for all the dates instead of one in case of overwrite. I'm not sure how I feel about that so I did not choose that route.

Reply:
I think one day at a time makes sense given the way Prefect maps things, but it depends a bit on the other question I asked- if we're ingesting multiple days of files then we should do multiple days of deletes in the same transaction. It would certainly be a lot faster to do in bulk, but the 99% case is just doing one date at a time.

Reply:
I think that the COPY INTO with a files argument has a limit of 1000 files?  It did for the Segment event pre-load.

Reply:
Yes, it has that 1000 limit, when you specify FILES=('file1', 'file2') etc.